src/fragments/lib/datastore/android/advanced-workflows/custom-primary-key/query-snippet.mdx

```diff
@@ -0,0 +1,113 @@
+With the value of the primary key:
+
+With a predicate:
+
+<Block name="Custom Primary Key">
+<BlockSwitcher>
+<Block name="Java">
+
+```java
+Amplify.DataStore.query(Book.class, Where.identifier(Book.class, "12345"),
+    matches -> {
+        if (matches.hasNext()) {
+            Book book = matches.next();
+            Log.i("MyAmplifyApp", "Title: " + book.getTitle());
+        }
+    },
+    failure -> Log.e("MyAmplifyApp", "Query failed.", failure));
+```
+
+</Block>
+<Block name="Kotlin - Callback">
+
+```kotlin
+Amplify.DataStore.query(Book::class.java, Where.identifier(Book::class.java, "12345"),
+    { allBooks: Iterator<Book> ->
+        while (allBooks.hasNext()) {
+            val book = allBooks.next()
+            Log.i("MyAmplifyApp", "Title: " + book.title)
+        }
+                            },
+   { failure: DataStoreException? -> Log.e("MyAmplifyApp", "Query failed.", failure) })
+```
+
+</Block>
+<Block name="Kotlin - Coroutines (Beta)">
+
+```kotlin
+Amplify.DataStore.query(Book::class, Where.identifier(Book::class.java, "12345"))
+    .catch { Log.e("MyAmplifyApp", "Query failed", it) }
+    .collect { Log.i("MyAmplifyApp", "Title: ${it.title}") }
+```
+
+</Block>
+<Block name="RxJava">
+
+```java
+RxAmplify.DataStore.query(Book.class, Where.identifier(Book::class.java, "12345")).subscribe(
+    book -> Log.i("MyAmplifyApp", "Title: " + book.getTitle()),
+    failure -> Log.e("MyAmplifyApp", "Query failed.", failure)
+);
+```
+
+</Block>
+</BlockSwitcher>
+</Block>
+<Block name="Composite Key">
+
```

Comment:
"With a predicate:" here i think

Reply:
@poojamat

Reply:
@lawmicha - confirmed with Pooja, the first block is for PK only, and the second block here should indeed be the composite key section.